external/fv3fit/fv3fit/emulation/losses.py

```diff
@@ -0,0 +1,82 @@
+import dataclasses
+from typing import List, Mapping
+
+import tensorflow as tf
+from fv3fit.emulation.layers.normalization import NormalizeConfig
+
+from .._shared.config import OptimizerConfig
+
+
+class NormalizedMSE(tf.keras.losses.MeanSquaredError):
+    """
+    Keras MSE that uses an emulation normalization class before
+    scoring
+    """
+
+    def __init__(self, norm_cls_name, sample_data, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._normalize = NormalizeConfig(norm_cls_name, sample_data).initialize_layer()
+
+    def call(self, y_true, y_pred):
+        return super().call(self._normalize(y_true), self._normalize(y_pred))
+
+
+@dataclasses.dataclass
+class CustomLoss:
+    """
+    Use custom custom normalized MSE-based losses for specified
+    variables
+
+    Args:
+        optimizer: configuration for the optimizer to
+            compile with the model
+        normalization: the normalization type (see normalization.py) to
+            use for the MSE
+        loss_variables: variable names to include in the MSE loss dict
+        metric_variables: variable names to include in the metrics dict
+        weights: custom scaling for the loss variables applied in the
+            overall keras "loss" term
+    """
+
+    optimizer: OptimizerConfig = dataclasses.field(
+        default_factory=lambda: OptimizerConfig("Adam")
+    )
```

Comment:
Remove optimizer from the dataclass

Reply:
I wanted to preserve backwards compatibility for now...so didn't change this interface. I'll probably separate the config from the `__call__`  implementation in a future PR.